tests/tests/test_api_management.py

```diff
@@ -178,3 +178,153 @@ def test_ok(self, tenant_id, api_client_mgmt, init_users_mt):
     @pytest.mark.parametrize("tenant_id", ["tenant1id", "tenant2id"])
     def test_no_users(self, tenant_id, api_client_mgmt, init_users_mt):
         self._do_test_no_users(api_client_mgmt, "non_existing_tenant_id")
+
+
+class TestManagementApiPutUserBase:
+    def _do_test_ok_email(self, api_client_mgmt, init_users, user, update, tenant_id=None):
+        auth = None
+        if tenant_id is not None:
+            auth = make_auth("foo", tenant_id)
+
+        # test update
+        _, r = api_client_mgmt.update_user(user.id, update, auth)
+        assert r.status_code == 204
+
+        # get/verify users
+        users = api_client_mgmt.get_users(auth)
+        assert len(users) == len(init_users)
+
+        found = [u for u in users if u.email == update["email"]]
+        assert len(found) == 1
+
+    def _do_test_ok_pass(self, api_client_mgmt, init_users, user, update, tenant_id=None):
+        auth = None
+        if tenant_id is not None:
+            auth = make_auth("foo", tenant_id)
+
+        # test update
+        _, r = api_client_mgmt.update_user(user.id, update, auth)
+        assert r.status_code == 204
+
+        # get/verify users
+        users = api_client_mgmt.get_users(auth)
+        assert len(users) == len(init_users)
+
+        found = [u for u in users if u.email == user.email]
+        assert len(found) == 1
+
+        # try if login still works
+        _, r = api_client_mgmt.login(user.email, update["password"])
+        assert r.status_code == 200
+
+    def _do_test_ok_email_and_pass(self, api_client_mgmt, init_users, user, update, tenant_id=None):
+        auth = None
+        if tenant_id is not None:
+            auth = make_auth("foo", tenant_id)
+
+        # test update
+        _, r = api_client_mgmt.update_user(user.id, update, auth)
+        assert r.status_code == 204
+
+        # get/verify users
+        users = api_client_mgmt.get_users(auth)
+        assert len(users) == len(init_users)
+
+        found = [u for u in users if u.email == update["email"]]
+        assert len(found) == 1
+
+        # try if login still works
+        _, r = api_client_mgmt.login(update["email"], update["password"])
+        assert r.status_code == 200
+
+    def _do_test_fail_not_found(self, api_client_mgmt, init_users, update, tenant_id=None):
+        auth = None
+        if tenant_id is not None:
+            auth = make_auth("foo", tenant_id)
+
+        try:
+            _, r = api_client_mgmt.update_user("madeupid", update, auth)
+        except bravado.exception.HTTPError as e:
+            assert e.response.status_code == 404
```

Comment:
yup, we control the init data set and 'madeupid' is dumb enough that we won't deliberately insert it to the db